benchmark/common.js

```diff
@@ -88,51 +89,28 @@ Benchmark.prototype._queue = function(options) {
   return queue;
 };
 
-function hasWrk() {
-  const result = child_process.spawnSync('wrk', ['-h']);
-  if (result.error && result.error.code === 'ENOENT') {
-    console.error('Couldn\'t locate `wrk` which is needed for running ' +
-      'benchmarks. Check benchmark/README.md for further instructions.');
-    process.exit(1);
-  }
-}
+// Benchmark an http server.
+exports.default_http_benchmarker =
+  http_benchmarkers.default_http_benchmarker;
+exports.supported_http_benchmarkers =
+  http_benchmarkers.supported_http_benchmarkers;
+exports.installed_http_benchmarkers =
+  http_benchmarkers.installed_http_benchmarkers;
 
-// benchmark an http server.
-const WRK_REGEXP = /Requests\/sec:[ \t]+([0-9\.]+)/;
-Benchmark.prototype.http = function(urlPath, args, cb) {
-  hasWrk();
+Benchmark.prototype.http = function(options, cb) {
   const self = this;
+  if (!options.port) {
+    options.port = exports.PORT;
+  }
 
-  const urlFull = 'http://127.0.0.1:' + exports.PORT + urlPath;
-  args = args.concat(urlFull);
-
-  const childStart = process.hrtime();
-  const child = child_process.spawn('wrk', args);
-  child.stderr.pipe(process.stderr);
-
-  // Collect stdout
-  let stdout = '';
-  child.stdout.on('data', (chunk) => stdout += chunk.toString());
-
-  child.once('close', function(code) {
-    const elapsed = process.hrtime(childStart);
-    if (cb) cb(code);
-
-    if (code) {
-      console.error('wrk failed with ' + code);
-      process.exit(code);
+  http_benchmarkers.run(options, function(benchmarker_name, result, elapsed) {
+    if (!self.config.benchmarker) {
+      self.config.benchmarker = benchmarker_name;
     }
-
-    // Extract requests pr second and check for odd results
-    const match = stdout.match(WRK_REGEXP);
-    if (!match || match.length <= 1) {
-      console.error('wrk produced strange output:');
-      console.error(stdout);
-      process.exit(1);
+    self.report(result, elapsed);
+    if (cb) {
+      cb(0);
```

Comment:
This is a little odd. I think you should make the `.run` signature

```
http_benchmarkers.run(options, function(error, results) { ... })
```

that way you can check the error and call the callback appropriately. Actually I don't care so much if the behaviour is maintained. It is just that we should avoid calling `process.exit(1)` from more than one file, as that makes the program difficult to reason about.